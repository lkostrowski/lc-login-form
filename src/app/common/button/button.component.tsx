import React, { HTMLAttributes } from 'react';
import cx from 'classnames';

import styles from './button.module.scss';

interface Props extends HTMLAttributes<HTMLButtonElement> {}

export const Button: React.FC<Props> = ({ className, ...props }) => {
    return <button className={cx(styles.button, className)} {...props} />;
};
