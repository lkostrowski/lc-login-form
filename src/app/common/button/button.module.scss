@import '../../../theme/colors';
@import '../../../theme/vars';

.button {
    background: linear-gradient(
        97.08deg,
        $blue 12.12%,
        rgba($violet, 0.49) 96.51%
    );
    height: 5rem;
    font-size: 1.6rem;
    color: $white;
    font-weight: 500;
    border: none;
    outline: none;
    border-radius: $borderRadius;
    cursor: pointer;
    text-align: left;
    position: relative;
    overflow: hidden;

    &::after {
        background: linear-gradient(
            97.08deg,
            $blue 5.12%,
            rgba($violet, 0.49) 66.51%
        );
        position: absolute;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        content: '';
        opacity: 0;
        transition: 0.2s;
    }

    &:hover {
        &::after {
            opacity: 1;
        }

        & .rightIconContainer {
            transform: translateY(-50%) translateX(0.5rem);
        }
    }
}

.content {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    display: flex;
    align-items: center;
    z-index: 1;
    padding: 0 2rem;
}

.fullWidth {
    display: block;
    width: 100%;
}

.withRightIcon {
    padding-right: 5rem;
}

.rightIconContainer {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    transition: 0.2s;
    right: 2rem;
}
