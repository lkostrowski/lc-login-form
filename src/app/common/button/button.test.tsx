import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Button } from './button.component';

afterEach(cleanup);

describe('Button component', () => {
    describe('Snapshots', () => {
        test('Basic version', () => {
            const { container } = render(<Button>Click me</Button>);

            expect(container).toMatchSnapshot();
        });
        test('With icon', () => {
            const { container } = render(
                <Button rightIcon={<svg />}>Click me</Button>,
            );

            expect(container).toMatchSnapshot();
        });
        test('Full width version', () => {
            const { container } = render(<Button fullWidth>Click me</Button>);

            expect(container).toMatchSnapshot();
        });
    });
});
