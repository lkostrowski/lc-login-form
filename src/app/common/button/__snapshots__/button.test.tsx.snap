// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Snapshots Basic version 1`] = `
<div>
  <button
    class="button"
  >
    <div
      class="content"
    >
      Click me
    </div>
  </button>
</div>
`;

exports[`Button component Snapshots Full width version 1`] = `
<div>
  <button
    class="button fullWidth"
  >
    <div
      class="content"
    >
      Click me
    </div>
  </button>
</div>
`;

exports[`Button component Snapshots With icon 1`] = `
<div>
  <button
    class="button"
  >
    <div
      class="content withRightIcon"
    >
      Click me
      <div
        class="rightIconContainer"
      >
        <svg />
      </div>
    </div>
  </button>
</div>
`;
